nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core-api/src/main/java/org/apache/nifi/controller/AbstractComponentNode.java

```diff
@@ -1084,12 +1092,29 @@ public void onParametersModified(final Map<String, ParameterUpdate> updatedParam
         }
 
         // If this component is affected by the Parameter change, we need to re-validate
+        componentAffected |= isConfigurationParameterModified(updatedParameters);
         if (componentAffected) {
             logger.debug("Configuration of {} changed due to an update to Parameter Context. Resetting validation state", this);
             resetValidationState();
         }
     }
 
+    protected void increaseReferenceCounts(final String parameterName) {
+        parameterReferenceCounts.merge(parameterName, 1, (a, b) -> a == -1 ? null : a + b);
+    }
+
+    protected void decreaseReferenceCounts(final String parameterName) {
```

Comment:
Given that this is always changing the value by `1`, it probably is best to call these `incrementReferenceCounts` and `decrementReferenceCounts`